sites/svelte-5-preview/src/routes/docs/content/01-api/05-imports.md

```diff
@@ -93,6 +93,37 @@ To prevent something from being treated as an `$effect`/`$derived` dependency, u
 </script>
 ```
 
+### `createRawSnippet`
+
+An advanced API designed for people building frameworks that integrate with Svelte, `createRawSnippet` allows you to create [snippets](/docs/snippets) programmatically for use with `{@render ...}` tags:
+
+```js
+import { createRawSnippet } from 'svelte';
+
+const greet = createRawSnippet((name) => {
+	return {
+		render: () => `
+			<h1>Hello ${name()}!</h1>
+		`,
+		setup: (h1) => {
+			$effect(() => {
+				h1.textContent = `Hello ${name()}!`;
+			});
```

Comment:
I imagine some of the uses for raw snippets are going to be people who want to pass in code that's _not_ going to be run through the Svelte compiler, right? Is there a way to make their lives easier, or are they just going to be responsible for manually updating the DOM whenever there's an update to something?

What does tearing down afterwards look like? I'm assuming any `$effect`s created in `setup` will have their teardown functions called when the snippet is destroyed, but how do people not using `$effect`s here find out that that is happening?

Reply:
We could maybe have an unmount function tbh. It would be useful maybe for libraries that don’t leverage out compiler.

Reply:
I think that could be as simple as calling a function returned from `setup` if there is one, right? I do think it is a good idea to provide that, because otherwise people writing these without `$effect`s are going to be writing code with memory leaks and no good way to avoid that.

Reply:
I mean that’s why you have the outer closure, no? You might want to cleanup the same thing  during render or setup.

Reply:
I think that'd be a real edge case, and if someone _does_ have that need they can do it all in the `render` function. Returning a teardown function from `setup` feels like the most ergonomic way to solve common cases

Reply:
Yeah that’s likely fine then